plugins/zenoh-plugin-storage-manager/src/replication/log.rs

```diff
@@ -224,22 +224,28 @@ impl LogLatest {
             return None;
         }
 
-        if let Ok((interval, sub_idx)) = self.configuration.get_time_classification(timestamp) {
-            if let Some(event) = self
-                .intervals
-                .get(&interval)
-                .and_then(|interval| interval.sub_interval_at(&sub_idx))
-                .and_then(|sub_interval| sub_interval.get(stripped_key))
-            {
-                return Some(event);
+        if let Ok((mut interval_idx, mut sub_idx)) =
+            self.configuration.get_time_classification(timestamp)
+        {
+            // Search forward from the current interval, subinterval
+            if let Some(interval) = self.intervals.get(&interval_idx) {
+                while let Some(sub_interval) = interval.sub_interval_at(&sub_idx) {
+                    if let Some(event) = sub_interval.get(stripped_key) {
+                        return Some(event);
+                    }
+                    sub_idx.0 = sub_idx.0 + 1;
+                }
             };
-        };
 
-        for interval in self.intervals.values().rev() {
-            if let Some(event) = interval.lookup(stripped_key) {
-                return Some(event);
+            // Search forward all intervals, including all Sub Intervals
+            interval_idx.0 = interval_idx.0 + 1;
+            while let Some(interval) = self.intervals.get(&interval_idx) {
+                if let Some(event) = interval.lookup(stripped_key) {
+                    return Some(event);
+                }
+                interval_idx.0 = interval_idx.0 + 1;
             }
```

Comment:
We have no guarantee that the intervals and sub-intervals are continuous, I think an iterator coupled with a filter is a viable approach.

For instance we might have:
```
- Interval(0)
   - SubInterval(0)
   - SubInterval(2)
- Interval(2)
   - SubInterval(1)
- Interval(4)
   - SubInterval(0)
   - SubInterval(2)
```